metricbeat/module/vsphere/host/host.go

```diff
@@ -86,89 +116,138 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
 		}
 
-		if hs.Summary.Host != nil {
-			networkNames, err := getNetworkNames(ctx, c, hs.Summary.Host.Reference())
-			if err != nil {
-				m.Logger().Debugf("error trying to get network names: %s", err.Error())
-			} else {
-				if len(networkNames) > 0 {
-					event["network_names"] = networkNames
-				}
+		results, err := perfManager.ToMetricSeries(ctx, samples)
+		if err != nil {
+			m.Logger().Errorf("Failed to convert performance data: %v", err)
+		}
+
+		metricMap := make(map[string]interface{})
+		for _, result := range results[0].Value {
+			if len(result.Value) > 0 {
+				metricMap[result.Name] = result.Value[0]
+				continue
 			}
+			m.Logger().Debugf("Metric %v: No result found", result.Name)
 		}
+
 		reporter.Event(mb.Event{
-			MetricSetFields: event,
+			MetricSetFields: m.eventMapping(hst[i], &metricData{
+				perfMetrics: metricMap,
+				assetsName:  *assetNames,
+			}),
 		})
 	}
 
 	return nil
 }
 
-func getNetworkNames(ctx context.Context, c *vim25.Client, ref types.ManagedObjectReference) ([]string, error) {
-	var outputNetworkNames []string
-
-	pc := property.DefaultCollector(c)
+func getAssetNames(ctx context.Context, pc *property.Collector, hs *mo.HostSystem) (*assetNames, error) {
+	referenceList := make([]types.ManagedObjectReference, 0, len(hs.Datastore)+len(hs.Vm))
+	referenceList = append(referenceList, hs.Datastore...)
+	referenceList = append(referenceList, hs.Vm...)
 
-	var hs mo.HostSystem
-	err := pc.RetrieveOne(ctx, ref, []string{"network"}, &hs)
-	if err != nil {
-		return nil, fmt.Errorf("error retrieving host information: %v", err)
-	}
-
-	if len(hs.Network) == 0 {
-		return nil, errors.New("no networks found")
-	}
-
-	var networkRefs []types.ManagedObjectReference
-	for _, obj := range hs.Network {
-		if obj.Type == "Network" {
-			networkRefs = append(networkRefs, obj)
+	var objects []mo.ManagedEntity
+	if len(referenceList) > 0 {
+		if err := pc.Retrieve(ctx, referenceList, []string{"name"}, &objects); err != nil {
+			return nil, err
 		}
 	}
 
-	if len(networkRefs) == 0 {
-		return nil, errors.New("no networks found")
+	outputDsNames := make([]string, 0, len(hs.Datastore))
+	outputVmNames := make([]string, 0, len(hs.Vm))
+	for _, ob := range objects {
+		name := strings.ReplaceAll(ob.Name, ".", "_")
+		switch ob.Reference().Type {
+		case "Datastore":
+			outputDsNames = append(outputDsNames, name)
+		case "VirtualMachine":
+			outputVmNames = append(outputVmNames, name)
+		}
 	}
 
-	var nets []mo.Network
-	err = pc.Retrieve(ctx, networkRefs, []string{"name"}, &nets)
-	if err != nil {
-		return nil, fmt.Errorf("error retrieving network from host: %v", err)
+	// calling network explicitly because of mo.Network's ManagedEntityObject.Name does not store Network name
+	// instead mo.Network.Name contains correct value of Network name
+	var netObjects []mo.Network
+	if len(hs.Network) > 0 {
+		if err := pc.Retrieve(ctx, hs.Network, []string{"name"}, &netObjects); err != nil {
+			return nil, err
+		}
```

Comment:
Declaring netObjects only makes sense if the condition `len(hs.Network) > 0` is met. 
```suggestion
	if len(hs.Network) > 0 {
	        var netObjects []mo.Network
		if err := pc.Retrieve(ctx, hs.Network, []string{"name"}, &netObjects); err != nil {
			return nil, err
		}
```